tests/integration/helpers.py

```diff
@@ -40,6 +42,43 @@ async def loki_rules(ops_test, app_name) -> dict:
         return {}
 
 
+async def get_alertmanager_alerts(ops_test: OpsTest, unit_name, unit_num, retries=3) -> List[dict]:
+    """Get a list of alerts.
+
+    Response looks like this:
+    {
+        'annotations': {'description': 'test-charm-...', 'summary': 'Instance test-charm-...'},
+        'endsAt': '2021-09-03T21:03:59.658Z',
+        'fingerprint': '4a0016cc12a07903',
+        'receivers': [{'name': 'pagerduty'}],
+        'startsAt': '2021-09-03T19:37:59.658Z',
+        'status': {'inhibitedBy': [], 'silencedBy': [], 'state': 'active'},
+        'updatedAt': '2021-09-03T20:59:59.660Z',
+        'generatorURL': 'http://prometheus-0:9090/...',
+        'labels': {
+            'alertname': 'AlwaysFiring',
+            'instance': 'test-charm-...',
+            'job': 'juju_test-charm-...',
+            'juju_application': 'tester', 'juju_model': 'test-charm-...',
+            'juju_model_uuid': '...',
+            'juju_unit': 'tester-0',
+            'severity': 'Low',
+            'status': 'testing'
+        }
+    }
+    """
+    address = await get_unit_address(ops_test, unit_name, unit_num)
+    url = f"http://{address}:9093/api/v2/alerts"
+    while not (alerts := json.loads(urllib.request.urlopen(url, data=None, timeout=2).read())):
+        retries -= 1
+        if retries > 0:
```

Comment:
ATTN @rbarry82 : I appreciate and like the simplicity of this approach. Thank you. I am going to do a couple of runs of this locally and if it works let us go with it. But if github runs start failing sporadically we could consider using exponential backoff here.